tests/rng_test.cpp

```diff
@@ -79,3 +79,57 @@ TEST_CASE( "random_entry_preserves_constness" )
     i1 = 5678;
     CHECK( v1[0] == 5678 );
 }
+
+template<int w, int h>
+static void test_pseudo_random_tiling_test( uint64_t seed )
+{
+    pseudo_random_matrix_tiling<w, h> mt( seed );
+    std::set<std::pair<int, int>> s1;
+    std::vector<std::pair<int, int>> vec1;
+
+    for( size_t i = 0; i < w * h; ++i ) {
+        const std::pair<int, int> &v = mt.get_xy( i );
+        CHECK( v.first >= 0 );
+        CHECK( v.first < w );
+        CHECK( v.second >= 0 );
+        CHECK( v.second <= h );
```

Comment:
Did you mean [0, h-1] here?
```suggestion
        CHECK( v.second < h );
```